tests/datatypes/test_str.py

```diff
@@ -240,6 +240,27 @@ def test_split(self):
                     print(err)
             """)
 
+    def test_splitlines(self):
+        self.assertCodeExecution("""
+            s = 'ab c\\nde fg\\rkl\\r\\nabc'
```

Comment:
I think more test cases are necessary to fully exercise the function – things like having two line boundaries in a row, having one or more line boundaries at the end of the string (which the example in the python docs had), empty string, and any other edge cases you can think of.